beacon-chain/core/helpers/validators.go

```diff
@@ -182,16 +184,18 @@ func BeaconProposerIndex(state *stateTrie.BeaconState) (uint64, error) {
 		if err != nil {
 			return 0, err
 		}
-		r, err := BlockRootAtSlot(state, s)
+		r, err := StateRootAtSlot(state, s)
 		if err != nil {
 			return 0, err
 		}
-		proposerIndices, err := proposerIndicesCache.ProposerIndices(bytesutil.ToBytes32(r))
-		if err != nil {
-			return 0, errors.Wrap(err, "could not interface with committee cache")
-		}
-		if proposerIndices != nil {
-			return proposerIndices[state.Slot()%params.BeaconConfig().SlotsPerEpoch], nil
+		if !bytes.Equal(r, params.BeaconConfig().ZeroHash[:]) {
+			proposerIndices, err := proposerIndicesCache.ProposerIndices(bytesutil.ToBytes32(r))
+			if err != nil {
+				return 0, errors.Wrap(err, "could not interface with committee cache")
+			}
+			if proposerIndices != nil {
+				return proposerIndices[state.Slot()%params.BeaconConfig().SlotsPerEpoch], nil
```

Comment:
Perhaps a length check here, just to be extra defensive even if it is unlikely to happen

Reply:
@rauljordan I think there is sufficient length check above. We know that len(proposerIndices) is exactly SlotsPerEpoch so when we do slot%SlotsPerEpoch, we know that there cannot be out of bounds since the max value of that operation would be the last value in the slice.